rss2email/main.py

```diff
@@ -163,25 +163,32 @@ def run(*args, **kwargs):
     if not getattr(args, 'func', None):
         parser.error('too few arguments')
 
-    lockfile_path = _os.path.join(_os.environ.get("XDG_RUNTIME_DIR",
-                                                  default="/tmp/"),
-                                  "rss2email.lock")
+    # Immediately lock so only one r2e instance runs at a time
+    if UNIX:
+        import fcntl as _fcntl
+        lockfile_path = _os.path.join(_os.environ.get("XDG_RUNTIME_DIR", default="/tmp/"), "rss2email.lock")
+        lockfile = open(lockfile_path, "w")
+        _fcntl.lockf(lockfile, _fcntl.LOCK_EX)
+        _LOG.debug("acquired lock file {}".format(lockfile_path))
+    else:
+        # TODO: What to do on Windows?
+        lockfile = None
+
     try:
-        with open(lockfile_path, "w") as lockfile:
-            # Immediately lock so only one r2e instance runs at a time
-            _fcntl.lockf(lockfile, _fcntl.LOCK_EX)
-            _LOG.debug("acquired lock file {}".format(lockfile_path))
-            if not args.config:
-                args.config = None
-            feeds = _feeds.Feeds(datafile_path=args.data, configfiles=args.config)
-            if args.func != _command.new:
-                feeds.load()
-            args.func(feeds=feeds, args=args)
+        if not args.config:
+            args.config = None
+        feeds = _feeds.Feeds(datafile_path=args.data, configfiles=args.config)
+        if args.func != _command.new:
+            feeds.load()
+        args.func(feeds=feeds, args=args)
     except _error.RSS2EmailError as e:
         e.log()
         if _logging.ERROR - 10 * args.verbose < _logging.DEBUG:
             raise  # don't mask the traceback
         _sys.exit(1)
+    finally:
+        if lockfile is not None:
```

Comment:
It is a bit of a shame that we have to do a try-finally instead of a `with`, but I guess this is needed to do nothing on Windows.

Reply:
I agree. We may introduce a helper context manager, e.g. `Lock`, and use `with Lock()` instead of `try`/`finally`. But I think this is out of scope of this commit.